pkg/pipeline/resource_pipeline_project_integration.go

```diff
@@ -282,3 +255,57 @@ func pipelineProjectIntegrationResource() *schema.Resource {
 		Description: "Provides an Jfrog Pipelines Project Integration resource.",
 	}
 }
+
+func unpackFormJSONValues(d *util.ResourceData, key string) []FormJSONValues {
+	var formJSONValues []FormJSONValues
+	keyValues := d.Get(key).([]interface{})
+	for _, keyValue := range keyValues {
+		idx := keyValue.(map[string]interface{})
+		formJSONValue := FormJSONValues{
+			Label: idx["label"].(string),
+			Value: idx["value"].(string),
+		}
+		formJSONValues = append(formJSONValues, formJSONValue)
+	}
+	return formJSONValues
+}
+
+func lookupFormJSONValue(values []FormJSONValues, key string) FormJSONValues {
```

Comment:
We've seen this func pattern in many places, so we added a generic func recently: https://github.com/jfrog/terraform-provider-artifactory/blob/master/pkg/artifactory/resource/configuration/configuration.go#L25

Reply:
Thanks! I will take a look and give that a try. 

Just reading the code, however, I'm not sure its exactly the same, at least for our use case. 

We do secret rotation with Terraform. So if the input GitHub API key (for example) changes from `secret_a` to `secret_b` we would like Pipelines to update the Integration value. If `secret_a` is not in the state, I am not sure offhand if that would pick up as a diff. 

I mostly based the idea off of the [AWS SSM provider](https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/ssm/parameter.go#L137), but I guess that is a slightly different use case and implementation on the API side.

Reply:
That does make sense, and I can see how its working in the Artifactory user schema. 

When trying here, if I skip packing that specific value inside of the `[]FormJSONValues` it never makes it to state.

If I skip packing `form_json_values` at all, it seems to work fine, and is a much simpler solution! I'll see if I can come up with a good enough unit test for that, but testing inside our actual project it seems perfect

Reply:
I think you'd still want to pack the `form_json_values` since there may be other values in that list that you need but are not sensitive.

One possibility is add a new boolean attribute (say `is_sensitive`) to the `form_json_values` TypeSet and then the packing code can decide whether to pack the value or not based on that.

Reply:
Something like:

```hcl
resource "pipeline_project_integration" "my-project-integration" {
  ...

  form_json_values {
    label = "some-sensitive-thing"
    value = "sensitive-value-that-artifactory-returns-redacted"
    is_sensitive = true
  }

  form_json_values {
    label = "label-2"
    value = "value-2"
  }
}
```

Reply:
Adding a new value is better/less hacky then checking the asterisks response. But I don't think it can be used as a conditional to pack a value in an array. 

The loop checking each `form_json_value` is creating a list of objects, and then outside that list there is a call to `setValue` with the entire list. In my testing, skipping any element of the list means the value is not committed to state, because we are explicitly removing it from the data array. If it was a normal map object I don't think its an issue, but because we are manipulating an array it will fail. 

Also I think we can't add a new value `is_sensitive` that is not returned by the API if we skip packing that element in the array, since that value will never be committed to state. Adding internal properties like that seem to only work when we can manipulate the values in pack/unpack.

Reply:
You're right in that to implement/add `is_sensitive` attribute for the form values, we will need to implement lower level un/packing function for each item of the list (vs setting the entire array into the attribute).

If you feel this will take more effort to implement than what you anticipated, feel free to create an issue and our team can work on this.

Reply:
Is there a different lower-level way other then using `d.Set("form_json_values.o.label", "my_label")`? 

That seems to be unsupported by the SDK https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/schema/field_writer_map.go#L72

```
2022/11/21 13:57:09 [ERROR] setting state: form_json_values.0.label: can only set full list
```

Reply:
Oh you're right. I forgot about that part. I still think it's possible to do but without diving way deeper into this, I can't say for sure. Let me know if you're planning to spend more effort investigating this or want us to take over.

Reply:
Yeah I am unsure what problem we are trying to solve with the implementation details of "packing the correct value" vs "skipping packing sensitive values". 

I'm happy to help a bit but not sure the difference between the proposed solution and the requested changes at this point.

Reply:
Circle back to original topic of this PR. If your testing convince you that using asterisks works well enough, I think we can proceed with this. I can explore different un/packing technique in a separate effort.

Reply:
I am pretty happy with the hybrid approach that is the current latest commit in the PR. Your idea for a sensitive flag is much better then relying on the magic string asterisks. It satisfies our testing and can be used in the future if you figure out a lower level packing/skip packing.